ord_schema/validations.py

```diff
@@ -443,19 +443,40 @@ def validate_addition_speed(message):
     del message  # Unused.
 
 
+def validate_amount(message):
+    if (message.HasField('volume_includes_solutes') and
+            message.WhichOneof('kind') != 'volume'):
+        warnings.warn(
+            'volume_includes_solutes should only be set for volume amounts',
+            ValidationError)
+
+
+def validate_source(message):
+    del message  # Unused.
+
+
 def validate_crude_component(message):
     if not message.reaction_id:
         warnings.warn('CrudeComponents must specify a reaction_id',
                       ValidationError)
-    if message.has_derived_amount and message.HasField('amount'):
+    if message.has_derived_amount and message.amount.HasField('kind'):
         warnings.warn(
             'CrudeComponents with derived amounts cannot have their'
             ' mass or volume specified explicitly', ValidationError)
     if ((not message.HasField('has_derived_amount') or
-         not message.has_derived_amount) and not message.HasField('amount')):
+         not message.has_derived_amount) and
+            not message.amount.HasField('kind')):
         warnings.warn(
             'Crude components should either have a derived amount or'
             ' a specified mass or volume', ValidationError)
+    if (message.amount.WhichOneof('kind') and
+            message.amount.WhichOneof('kind') not in ['mass', 'volume']):
+        warnings.warn(
+            'Crude component amounts must be specified by mass or volume',
+            ValidationError)
+    if message.amount.HasField('volume_includes_solutes'):
+        warnings.warn(
+            'volume_includes_solutes should only be used for input Compounds')
```

Comment:
Missing `ValidationError` argument to `warnings.warn`

Reply:
Done.